image/tarexport/save.go

```diff
@@ -73,28 +69,36 @@ func (l *tarexporter) parseNames(names []string) (map[image.ID]*imageDescriptor,
 	}
 
 	for _, name := range names {
-		id, ref, err := reference.ParseIDOrReference(name)
+		ref, err := reference.ParseAnyReference(name)
 		if err != nil {
 			return nil, err
 		}
-		if id != "" {
-			_, err := l.is.Get(image.IDFromDigest(id))
-			if err != nil {
-				return nil, err
+		namedRef, ok := ref.(reference.Named)
+		if !ok {
+			// Check if digest ID reference
+			if digested, ok := ref.(reference.Digested); ok {
+				id := image.IDFromDigest(digested.Digest())
+				_, err := l.is.Get(id)
+				if err != nil {
+					return nil, err
+				}
+				addAssoc(id, nil)
+				continue
+			} else {
+				return nil, errors.Errorf("invalid reference: %v", name)
 			}
-			addAssoc(image.IDFromDigest(id), nil)
-			continue
 		}
-		if ref.Name() == string(digest.Canonical) {
+
+		if reference.FamiliarName(namedRef) == string(digest.Canonical) {
```

Comment:
Oh. I forgot we support this. It seems a bit strange to combine a `sha256:` prefix with a truncated digest, but we probably had a good reason to make it work.

> This should probably be updated in the future to use https://godoc.org/github.com/docker/distribution/reference#ParseAnyReferenceWithSet instead

Yes please!